app/models/user.rb

```diff
@@ -456,6 +456,16 @@ def close_account!
     AccountDeletion.create(:person => self.person)
   end
 
+  def lock_account!
+    self.locked_at = Time.now
+    self.save
+  end
```

Comment:
Devise is an authentication framework, a library that handles user authentication. And we use it, you can see that in this verify file, notice the `devise` at the top.